hazelcast/src/main/java/com/hazelcast/sql/impl/operation/QueryOperationHandlerImpl.java

```diff
@@ -131,31 +145,18 @@ public boolean submitRemote(UUID callerId, Connection connection, QueryOperation
         }
     }
 
-    @Override
-    public QueryOperationChannel createChannel(UUID sourceMemberId, UUID targetMemberId) {
-        if (targetMemberId.equals(getLocalMemberId())) {
-            return new QueryOperationChannelImpl(this, sourceMemberId, null);
-        } else {
-            Connection connection = getConnection(targetMemberId);
-
-            if (connection == null) {
-                throw QueryException.memberConnection(targetMemberId);
-            }
-
-            return new QueryOperationChannelImpl(this, sourceMemberId, connection);
-        }
-    }
-
     @Override
     public void execute(QueryOperation operation) {
+        assert operation.isSystem() == QueryOperationWorkerPool.isSystemThread();
+
         if (operation instanceof QueryExecuteOperation) {
             handleExecute((QueryExecuteOperation) operation);
-        } else if (operation instanceof QueryBatchExchangeOperation) {
-            handleBatch((QueryBatchExchangeOperation) operation);
+        } else if (operation instanceof QueryExecuteFragmentOperation) {
```

Comment:
Just a way to organize the code. Currently, messages are POJOs, and the logic is concentrated in the `QueryOperationHandlerImpl`. Otherwise, we would have to pass some context/state to messages, which may make the code a little bit more complicated. But in any case, this is primarily about styling. Both approaches works.